File: terre09.js
// Nombre premier
const args = process.argv.slice(2)
if (args.length != 1) {
    console.log("Usage: terre09.js <number>")
    process.exit(1)
}

const number = parseInt(args[0])

if (isNaN(number)) {
    console.log(`Expected a number got=${arg}`)
    process.exit(1)
}

if (number < 2) {
    console.log(`Non, ${number} n'est pas un nombre premier.`)
} else {
    for (let i = 2; i < number / 2; i++) {
        if (number % i == 0) {
            console.log(`Non, ${number} n'est pas un nombre premier. ${number}/${i}=${number / i}`)
            process.exit(0)
        }
    }
    console.log(`Oui, ${number} est un nombre premier.`)
}

File: terre09.exs
# Nombre Premier

arg =
  System.argv()
  |> case do
    [arg] ->
      arg

    _ ->
      IO.puts("Usage: elixir terre09.exs <number>")
      System.stop(1)
  end

case Integer.parse(arg) do
  :error ->
    IO.puts("Expected a valid number got=#{arg}")
    System.stop(1)

  {number, _rest} when number < 2 ->
    IO.puts("Non, #{number} n'est pas un nombre premier.")

  {number, _rest} ->
    divisible_by =
      2..round(number / 2)
      |> Enum.find(fn i ->
        rem(number, i) == 0
      end)

    if divisible_by == nil do
      IO.puts("Oui, #{number} est un nombre premier.")
    else
      division_result = round(number / divisible_by)

      IO.puts(
        "Non, #{number} n'est pas un nombre premier. #{number}/#{divisible_by}=#{division_result}"
      )
    end
end
